stat/distmat/permutation.go

```diff
@@ -0,0 +1,49 @@
+// Copyright ©2020 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package distmat
+
+import (
+	"golang.org/x/exp/rand"
+
+	"gonum.org/v1/gonum/mat"
+)
+
+// UniformPermutation is a uniform distribution over the n!
+// permutation matrices of size n x n for a given n.
+type UniformPermutation struct {
+	rnd     *rand.Rand
+	indices []int
+}
+
+// NewUniformPermutation constructs a new permutation matrix
+// generator using the given random source.
+func NewUniformPermutation(src rand.Source) *UniformPermutation {
+	return &UniformPermutation{rnd: rand.New(src)}
+}
+
+// PermTo sets the given matrix to be a random permutation matrix.
+// It does not zero the destination's elements, so it is the responsibility
+// of the caller to ensure it is correctly conditioned prior to the call.
+//
+// PermTo panics if m is not square.
```

Comment:
```
// PermTo sets the elements of a random permutation matrix into dst.
// It does not zero the destination's elements, so it is the responsibility
// of the caller to ensure it is correctly conditioned prior to the call.
//
// PermTo panics if dst is not square.
```

I'm not convinced about the first sentence's language yet, but I can't think of better wording yet.